Examples/Physics_applications/capacitive_discharge/PICMI_inputs_1d.py

```diff
@@ -0,0 +1,377 @@
+
+        self.nxguardphi = 1
+        self.nzguardphi = 1
+
+        self.phi = np.zeros(self.nz + 1 + 2*self.nzguardphi)
+
+        self.decompose_matrix()
+
+        callbacks.installpoissonsolver(self._run_solve)
+
+    def decompose_matrix(self):
+        """Function to build the superLU object used to solve the linear
+        system."""
+        self.nsolve = self.nz + 1
+
+        # Set up the tridiagonal computation matrix in order to solve A*phi =
+        # rho for phi.
+        self.A_ldiag = np.ones(self.nsolve-1) / self.dz**2
+        self.A_mdiag = -2.*np.ones(self.nsolve) / self.dz**2
+        self.A_udiag = np.ones(self.nsolve-1) / self.dz**2
+
+        self.A_mdiag[0] = 1.
+        self.A_udiag[0] = 0.0
+
+        self.A_mdiag[-1] = 1.
+        self.A_ldiag[-1] = 0.0
+
+        # Set up the computation matrix in order to solve A*phi = rho
+        A = np.zeros((self.nsolve, self.nsolve))
+
+        for ii in range(self.nsolve):
+            for jj in range(self.nsolve):
+                if ii == jj:
+                    A[ii, jj] = -2.0
+                elif ii == jj - 1 or ii == jj + 1:
+                    A[ii, jj] = 1.0
+
+        A[0, 1] = 0.0
+        A[-1, -2] = 0.0
+        A[0, 0] = 1.0
+        A[-1, -1] = 1.0
+
+        A = csc_matrix(A, dtype=np.float32)
+        self.lu = sla.splu(A)
+
+    def _run_solve(self):
+        """Function run on every step to perform the required steps to solve
+        Poisson's equation."""
+
+        # get rho from WarpX
+        if self.rho_wrapper is None:
+            self.rho_wrapper = fields.RhoFPWrapper(0, False)
+        self.rho_data = self.rho_wrapper[Ellipsis][:,0]
+
+        # run superLU solver to get phi
+        self.solve()
+
+        # write phi to WarpX
+        if self.phi_wrapper is None:
+            self.phi_wrapper = fields.PhiFPWrapper(0, True)
+        self.phi_wrapper[Ellipsis] = self.phi
+
+    def solve(self):
+        """The solution step. Includes getting the boundary potentials and
+        calculating phi from rho."""
+
+        left_voltage = 0.0
+        right_voltage = eval(
+            self.right_voltage,
+            {'t':self.sim.extension.gett_new(0), 'sin':np.sin, 'pi':np.pi}
+        )
+
+        # Construct b vector
+        rho = -self.rho_data / constants.ep0
+        b = np.zeros(rho.shape[0], dtype=np.float32)
+        b[:] = rho * self.dz**2
+
+        b[0] = left_voltage
+        b[-1] = right_voltage
+
+        phi = self.lu.solve(b)
+
+        self.phi[self.nzguardphi:-self.nzguardphi] = phi
+
+        self.phi[:self.nzguardphi] = left_voltage
+        self.phi[-self.nzguardphi:] = right_voltage
+
+
+class CapacitiveDischargeExample(object):
+
+    #######################################################################
+    # Begin global user parameters                                        #
+    #######################################################################
+
+    D_CA = 0.067 # m
```

Comment:
Can a description of this system be added here, describing what is being simulated and the meaning of each of the parameters? What is the purpose of the four different cases?

Note that the PEP8 standard is to use lower case for variable names.

Reply:
I've added a bit more of a description and changed some variable names to be more self explanatory. The upper case variable names is a practice we use to distinguish user input parameters from derived quantities but I have changed them back to lower case in this input file.